### DISCUSSION TITLE
Training/Predicting with 2 Images as Input

### DISCUSSION THREAD

**Discussion by EdwardAndersonMcDermott** [ultralytics/yolov5#7684] [Q&A]
I'm curious how much work it would be to adjust the algorithm to work with 2 images rather than 1.  I.e. a standard RGB image plus a thermal RGB image.  This would make inputs [6, 640, 640] as apposed to [3, 640, 640].

In principle how could I handle an arbitrary number of colour channels?

**Comment by glenn-jocher** [comment]
@EdwardAndersonMcDermott arbitrary channel inputs are a common feature request. We have this in our roadmap for Q2 2022. We haven't started yet, so if you'd like to contribute, the main modifications need to be done in the dataloaders. 

The models themselves can already be constructed with any channels by passing ch=6 in your model.yaml.

Please see our ✅ [Contributing Guide](https://github.com/ultralytics/yolov5/blob/master/CONTRIBUTING.md) to get started.

**Comment by EdwardAndersonMcDermott** [comment]
Ok, thanks @glenn-jocher.  Sounds like a reasonable thing then so I'll add it to my workload and see if I can contribute.

**Reply by kristeleg** [reply]
@EdwardAndersonMcDermott Did you manage to modify the dataloaders and successfully train on 2 images? 
I am trying to train on 2 images as well but it has not been very straightforward.

**Reply by EdwardAndersonMcDermott** [reply]
@kristinatel I've not been able to attempt this yet sorry.

**Reply by glenn-jocher** [reply]
@kristinatel @EdwardAndersonMcDermott arbitrary dataset sizes are supported down to 1 image.
